aspects/aspects.go

```diff
@@ -128,8 +128,39 @@ type DataBag interface {
 // be committed.
 type Schema interface {
 	Validate(data []byte) error
+
+	// AtPath returns the schemas (e.g., string, int, etc) that may be at the
+	// provided path. If the path cannot be followed, an error is returned.
+	AtPath(path []string) ([]Schema, error)
+
+	// Type returns the SchemaType corresponding to the Schema or Invalid if
+	// the Schema implementation is not one of the fundamental types.
+	Type() SchemaType
+}
+
+type SchemaType uint
+
+func (v SchemaType) String() string {
+	if int(v) >= len(typeStrings) {
+		panic("unknown schema type")
+	}
+
+	return typeStrings[v]
 }
 
+const (
+	Invalid SchemaType = iota
```

Comment:
I'm not completely sold on having this, maybe we can have a type of alternative instead